spring-cloud-starter-stream-source-websocket/src/main/java/org/springframework/cloud/stream/app/websocket/source/WebsocketSourceConfiguration.java

```diff
@@ -0,0 +1,40 @@
+package org.springframework.cloud.stream.app.websocket.source;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.websocket.IntegrationWebSocketContainer;
+import org.springframework.integration.websocket.ServerWebSocketContainer;
+import org.springframework.integration.websocket.inbound.WebSocketInboundChannelAdapter;
+
+/**
+ * A source module that receives data over WebSocket
+ *
+ * @author krishnaprasad
+ *
+ */
+@Configuration
+@EnableConfigurationProperties(WebsocketSourceProperties.class)
+@EnableBinding(Source.class)
+public class WebsocketSourceConfiguration {
+
+	@Autowired
+	private WebsocketSourceProperties properties;
+
+	@Bean
+	public WebSocketInboundChannelAdapter webSocketInboundChannelAdapter() {
+		WebSocketInboundChannelAdapter webSocketInboundChannelAdapter = new WebSocketInboundChannelAdapter(
+				serverWebSocketContainer());
+		webSocketInboundChannelAdapter.setOutputChannelName(Source.OUTPUT);
+		return webSocketInboundChannelAdapter;
+	}
+
+	@Bean
+	public IntegrationWebSocketContainer serverWebSocketContainer() {
+		return new ServerWebSocketContainer(properties.getPath()).setAllowedOrigins("*");
```

Comment:
We should add `Allowed Origins` as a configurable property

Reply:
Thank you @viniciusccarvalho for the  comments. Definitely I will fix the issues in your guidance.
About the Readme I'm getting the following error,

[ERROR] Failed to execute goal org.springframework.cloud:spring-cloud-app-starter-doc-maven-plugin:1.1.4.RC1:generate-documentation (generate-documentation) on project spring-cloud-starter-stream-source-websocket: Error generating documentation: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.springframework.cloud:spring-cloud-app-starter-doc-maven-plugin:1.1.4.RC1:generate-documentation (generate-documentation) on project spring-cloud-starter-stream-source-websocket: Error generating documentation
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:216)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
        at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
        at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:50)
Caused by: org.apache.maven.plugin.MojoExecutionException: Error generating documentation
        at org.springframework.cloud.stream.app.documentation.plugin.ConfigurationMetadataDocumentationMojo.execute(ConfigurationMetadataDocumentationMojo.java:138)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 27 more
Caused by: java.lang.NullPointerException
        at org.springframework.cloud.stream.app.documentation.plugin.ConfigurationMetadataDocumentationMojo.maybeHints(ConfigurationMetadataDocumentationMojo.java:175)
        at org.springframework.cloud.stream.app.documentation.plugin.ConfigurationMetadataDocumentationMojo.asciidocFor(ConfigurationMetadataDocumentationMojo.java:159)
        at org.springframework.cloud.stream.app.documentation.plugin.ConfigurationMetadataDocumentationMojo.execute(ConfigurationMetadataDocumentationMojo.java:119)
        ... 29 more

Can you help me to find out the mistake I have done?